### PR TITLE
LPS-184945 [Site Initializer] Link Commerce Specification Groups to Labels

### PR DESCRIPTION
**Pull request by joanagrego**

### CODE REVIEW

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+		"cpDefinitionExternalReferenceCode": "TESTCOMMERCEPRODUCT1",
+		"description": "",
+		"facetable": false,
+		"key": "test-product-specification-1",
+		"productSpecificationValue": {
+			"en_US": "1"
+		},
+		"title": "Test Product Specification 1 update"
+	}
+]
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json:13]: lets add more specifications
like catalog we can put 2 specification in bundle-1 and 3 in bundle-2

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	{
+		"categoryKey": "test-commerce-specification-key-1",
+		"cpDefinitionExternalReferenceCode": "TESTCOMMERCEPRODUCT1",
+		"description": "",
+		"facetable": false,
+		"key": "test-product-specification-1",
+		"productSpecificationValue": {
+			"en_US": "1"
+		},
+		"title": "Test Product Specification 1 update"
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json:11]:
```suggestion
		"title": "Test Product Specification 1 Update"
```

**Comment by joanagrego** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 3]
```diff
@@ -721,40 +749,43 @@ private void _assertCommerceNotificationTemplate(
...
 	}
 
-	private void _assertCommerceSpecificationProducts() throws Exception {
+	private void _assertCommerceSpecificationProducts1() throws Exception {
 		CPSpecificationOption cpSpecificationOption =
 			_cpSpecificationOptionLocalService.fetchCPSpecificationOption(
 				_serviceContext.getCompanyId(), "test-product-specification-1");
 
 		Assert.assertNotNull(cpSpecificationOption);
+		Assert.assertFalse(cpSpecificationOption.getCPOptionCategoryId() > 0);
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:758]: let's also assert the name of specification to check if it's updated

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		"facetable": false,
+		"key": "test-product-specification-2",
+		"productSpecificationValue": {
+			"en_US": "1"
+		},
+		"title": "Test Product Specification 2 Update"
+	},
+	{
+		"categoryKey": "test-commerce-specification-key-1",
+		"cpDefinitionExternalReferenceCode": "TESTCOMMERCEPRODUCT3",
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json:26]: link to a existing product (TESTCOMMERCEPRODUCT1) or remove

**Review by nNilton — CHANGES_REQUESTED** [review]

**Comment by joanagrego** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	{
+		"categoryKey": "test-commerce-specification-key-1",
+		"cpDefinitionExternalReferenceCode": "TESTCOMMERCEPRODUCT1",
+		"description": "",
+		"facetable": false,
+		"key": "test-product-specification-3",
+		"productSpecificationValue": {
+			"en_US": "1"
+		},
+		"title": "Test Product Specification 3"
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json:33]:
```suggestion
		"title": "Test Commerce Product Specification 3"
```

Comment (nNilton) [reply modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-2/src/main/resources/site-initializer/commerce-catalogs/test-commerce-catalog-1.products.specifications.json:33]: for all of them
and same for the key test-commerce-product-specification

**Review by nNilton — CHANGES_REQUESTED** [review]

**Review by nNilton — APPROVED** [review]

**Comment by nNilton** [comment]
ci:forward

**Comment by joseabelenda** [comment]
resent:https://github.com/pt-liferay-solutions/liferay-portal/pull/4125

**Pull request closed without merging**